tools/clang/tools/dxr/dxr.cpp

```diff
@@ -35,7 +35,10 @@ using namespace hlsl::options;
 #ifdef _WIN32
 int __cdecl wmain(int argc, const wchar_t **argv_) {
 #else
-int main(int argc, const char **argv_) {
+int main(int argc, const char **argv) {
+  // Convert argv to wchar.
+  WArgV ArgV(argc, argv);
+  const wchar_t **argv_ = ArgV.argv();
```

Comment:
This is a little icky where the previous code followed the same flow for all platforms. If it's needed or even if it's just a temporary fix, that's fine, but https://github.com/microsoft/DirectXShaderCompiler/pull/4895/files#r1055802917 indicates it was done this way initially to avoid these very ifdefs

Reply:
This ickiness is amplified by the fact that this is the only place that WargV is used.

`Dxclib.cpp` parses args, then uses `CopyArgsToWStrings` to copy parsed args to wchar_t strings for use in the API.  Could this code follow the same pattern using `RewriteOption` in place of `CoreOption`?  Then we wouldn't even need this `WargV` adapter, right?

Or perhaps rewriter functionality should just be integrated into dxc so we can just delete this code.

Reply:
I guess I don't feel too strongly about it.  For now, I'm ok with this if @pow2clk is.

Reply:
I don't personally object to the code. I wanted to ensure we had the discussion and I feel we have